sycl/source/detail/event_impl.hpp

```diff
@@ -416,6 +397,20 @@ class event_impl {
                   std::shared_ptr<sycl::detail::context_impl> Context);
 
   std::atomic_bool MIsEnqueued{false};
+
+  // Events constructed without a context will lazily use the default context
+  // when needed.
+  void tryToInitContext();
+  // Event class represents 3 different kinds of operations:
+  // | type  | has PI event | MContext | MIsHostTask | MIsDefaultConstructed |
```

Comment:
`MIsHostTask` should be `MIsHostEvent` no?

Reply:
I wanted to be more specific here. Currently "host" event should exist only if it was attached to host task. No other case when "host" event could and should be created.